### DISCUSSION TITLE
Is it possible to use {{ID}} of user form live chat?

### DISCUSSION THREAD

**Discussion by b4zz4** [JohnXLivingston/peertube-plugin-livechat#61] [Ideas]
For example: "user@room.xmpp.server.tv" to unify all chats of a user

**Comment by b4zz4** [comment]
room = room.replace(/{{VIDEO_UUID}}/g, video.uuid)
room = room.replace(/{{VIDEO_CHANNELID}}/g, video.channelId)

**Comment by JohnXLivingston** [comment]
This was asked here: https://github.com/JohnXLivingston/peertube-plugin-livechat/issues/58
I already added this to my roadmap: https://github.com/JohnXLivingston/peertube-plugin-livechat/blob/main/ROADMAP.md

Before accepting your pull request (https://github.com/JohnXLivingston/peertube-plugin-livechat/pull/62), i'd like to find a way to do this with the «builtin Prosody» mode (is that an instance wide settings? per video? etc...).
I will come back to you soon.

**Reply by JohnXLivingston** [reply]
Available with this commit: https://github.com/JohnXLivingston/peertube-plugin-livechat/commit/a3579abb8160cfdba192be106e15033bbd826616

**Reply by JohnXLivingston** [reply]
Full feature is released (v4.0.0)

**Comment by b4zz4** [comment]
Maybe it'd be good too: {{USER_ID}}

**Reply by JohnXLivingston** [reply]
It is more difficult, as we don't have any user information in the video object. It would require an extra request (sql for backend or api from frontend, depending on the chat mode). I tried, and when I realized it was not so easy I dropped the changes. Channel seems enough for me.
